include/taco/lower/lowerer_impl_C.h

```diff
@@ -0,0 +1,530 @@
+#ifndef TACO_LOWERER_IMPL_C_H
+#define TACO_LOWERER_IMPL_C_H
+
+#include <utility>
+#include <vector>
+#include <map>
+#include <set>
+#include <memory>
+#include <taco/index_notation/index_notation.h>
+
+#include "taco/lower/iterator.h"
+#include "taco/util/scopedset.h"
+#include "taco/util/uncopyable.h"
+#include "taco/ir_tags.h"
+#include "taco/lower/lowerer_impl.h"
+
+namespace taco {
+
+class TensorVar;
+class IndexVar;
+
+class IndexStmt;
+class Assignment;
+class Yield;
+class Forall;
+class Where;
+class Multi;
+class SuchThat;
+class Sequence;
+
+class IndexExpr;
+class Access;
+class Literal;
+class Neg;
+class Add;
+class Sub;
+class Mul;
+class Div;
+class Sqrt;
+class Cast;
+class CallIntrinsic;
+
+class MergeLattice;
+class MergePoint;
+class ModeAccess;
+
+namespace ir {
+class Stmt;
+class Expr;
+}
+
+class LowererImplC : public LowererImpl {
```

Comment:
`LowererImplImperative` (or something to that effect) might be a more accurate name than `LowererImplC`, since the generated IR can in turn be targeted to other imperative languages (e.g., LLVM IR)?

Reply:
Ok I can change that too. I just used LowererImplC in my spatial branch so that's what I used here

Reply:
Changed 777519d